modules/host/network.go

```diff
@@ -18,14 +18,14 @@ var rpcSettingsDeprecated = types.Specifier{'S', 'e', 't', 't', 'i', 'n', 'g', '
 // checks if the host's hostname has changed, and makes an updated host
 // announcement if so.
 func (h *Host) threadedUpdateHostname() {
+	err := h.tg.Add()
+	if err != nil {
+		return
+	}
+	defer h.tg.Done()
+
 	for {
-		err := h.tg.Add()
-		if err != nil {
-			return
-		}
 		h.managedLearnHostname()
-		h.tg.Done()
-
```

Comment:
It was more significant when the host was using resource locks instead of thread groups, because there was no shutdown chan meaning that the lock could be held for up to thirty minutes in a row. That was definitely important to fix, but the `stopChan` makes it okay for the lock grab to be long-running again.